e2e_tests/tests/per_provider/normal_tests/hash.rs

```diff
@@ -35,10 +35,17 @@ fn hash_not_supported() {
     }
 
     if !client.is_operation_supported(Opcode::PsaHashCompare) {
+        #[cfg(not(feature = "cryptoauthlib-provider"))]
         assert_eq!(
             client.hash_compare(Hash::Sha256, &[], &[]).unwrap_err(),
             ResponseStatus::PsaErrorNotSupported
         );
+        // CryptoAuthlib Provider checks arguments first and finds the hash to be too short.
+        #[cfg(feature = "cryptoauthlib-provider")]
+        assert_eq!(
+            client.hash_compare(Hash::Sha256, &[], &[]).unwrap_err(),
+            ResponseStatus::PsaErrorInvalidArgument
+        );
```

Comment:
Things are getting "trickier". The CryptoAuthLib provider may implement more callbacks then reported opcodes - at the moment this is simple difference between ATECC608 and test-interface, but in the future e.g. ATSHA208 can also be addressed.
Here it looks like Parsec calls the callback even though the opcode is missing. Should it be protected by the provider, by means of adding a protection at the beginning of the callback? Or on the Parsec level, i.e. do not call callbacks not backed by opcodes? If the latter, could you please help me finding where it can be done in the code?

Reply:
I think providers should be "self-contained"/"self-consistent", i.e. if they report that an opcode isn't available, they should also reply with `PsaErrorNotSupported` to that operation (mostly because the rest of the system isn't really intended to understand operations, only to pass things back and forth). You could probably check your pre-generated list of opcodes at the start of the call

Reply:
I will add this check, but in general this is an overkill - there may be something like 40 checks (5 providers x 8 operations per provider) to be added. On the other side, there is only CALib provider that needs to do it, at least at the moment.

Reply:
For the other providers it isn't needed because we can map the response code from the backend to a "not-supported" - is that possile for CAL? Should've been my first questiion

Reply:
OK - so the actual fix for this is here: ~/parsec/src/providers/cryptoauthlib/mod.rs. Just pushed together with other fixes.

Reply:
Cheers! I assume that most of these could be removed when they'll be supported by the mock implementation too, only keeping the ones that vary by backend.